Documentation/General/productPlanning/planning.tex

```diff
@@ -0,0 +1,154 @@
+
+\begin{itemize}
+\item Assess the quality of the initial algorithm.
+\item Incorporate user feedback in the algorithm.
+\end{itemize}
+
+\item Week 6: Sprint 5\\
+The music-matching algorithm will be tweaked and extended. We also review our own code and test quality, since we're half way.
+
+\begin{itemize}
+\item Review code and test quality.
+\item Improve Algorithm
+\end{itemize}
+
+\item Week 7: Sprint 6\\
+We will apply Sig feedback and implement an auto room creation system.
+
+\begin{itemize}
+\item Apply Sig feedback.
+\item Implement auto room creation system.
+\end{itemize}
+
+\item Week 8: Sprint 7\\
+Finish documentation and start final report.
+
+\begin{itemize}
+\item Finish architecture design document.
+\item Create final report.
+\end{itemize}
+
+
+\item Week 9: Sprint 8\\
+Finish final report and prepare presentation.
+
+\begin{itemize}
+\item Finish final report.
+\item Prepare and practise presentation.
+\end{itemize}
+
+
+\end{itemize}
+
+\chapter{Product backlog}
+
+\section{User stories of features}
+-As a user,\\
+-When I start the app,\\
+-Then I should see a mood-selection page.
+-
+-As a user,\\
+-When I select a mood,\\
+-Then I should see a list of rooms.
+-
+-As a user,\\
+-When I select a room,\\
+-Then the current song should be played.
+-
+-As a user,\\
+-When I select a room,\\
+-Then I should see a chat box.
+-
+-As a user,\\
+-When I type in the chat message input field,\\
+-And I click on 'Send message',\\
+-Then the message should be send to the backend.
+-
+-As a user,\\
+-When a user in my current room sends a message to the backend,\\
+-Then I should see that message in the chat box.
+-
+-As a user,\\
+-When I click on the 'thumbs up' or 'thumbs down' buttons,\\
+-Then the system should alter according to the feedback to enhance the song-selection algorithm.
+-
+-As a user,\\
+-When I click on the 'mute' button,\\
+-Then the music should be muted.
+-
+-As a user,\\
+-When I join a room,\\
+-Then I should see the latest messages sent in this room.
+-
+\section{User stories of defects (if applicable)}
+As of the time of writing, no defects have been discovered.
+
+\section{User stories of technical improvements (if applicable)}
+As of the time of writing, no technical improvements were made.
+
+\section{User stories of know-how acquisition}
+
+\section{Initial release plan}
+
+
+\chapter{Definition of Done}
+We need to set a definition of done.
+This entails that for every feature, sprint and the end product, we will define what we consider "done".\\
+
+A feature is done when it is unit tested to such an extend that our set coverage is reached. This implies the feature works as defined. Code written for the feature must also be documented and follow our checkstyle and markup rules. Finally, the feature must be reviewed by at least one team member that was not directly involved in creating the code and tests. If all these prerequisites are met, the Pull-Request can be merged.\\
```

Comment:
I would go for
> A feature is completed when it is implemented according to its specification and the code is well tested using unit tests. Furthermore, the public API (the public methods) of introduced classes should be well-documented using JavaDoc or NgDoc.
Coverage is a requirement, but we are not testing for coverage, we're testing functionality. Having documentation for the code is actually required in the process, so we should mention it.